app/src/main/java/com/infomaniak/mail/ui/main/move/MoveAdapter.kt

```diff
@@ -178,35 +175,20 @@ class MoveAdapter @Inject constructor(
         }
     }
 
-    @SuppressLint("NotifyDataSetChanged")
     fun setFolders(newFolders: List<Folder>, newCurrentFolderId: String? = null) = runCatchingRealm {
-
-        foldersDiffer.submitList(newFolders)
-
-        setFoldersJob?.cancel()
-        setFoldersJob = globalCoroutineScope.launch {
-            newCurrentFolderId?.let { currentFolderId = it }
-            val newHasCollapsableFolder = newFolders.any { it.canBeCollapsed }
-
-            val isFirstTime = hasCollapsableFolder == null
-            val collapsableFolderExistenceHasChanged = newHasCollapsableFolder != hasCollapsableFolder
-            hasCollapsableFolder = newHasCollapsableFolder
-
-            if (!isFirstTime && collapsableFolderExistenceHasChanged) {
-                Dispatchers.Main { notifyDataSetChanged() }
-            }
-        }
+        newCurrentFolderId?.let { currentFolderId = it }
```

Comment:
Do we really need to make `newCurrentFolderId` nullable and give it a default value?

Reply:
How do you want to do ?

Reply:
Yeah, I don't know what I read the first time around, the value still needs to be nullable. Still, wouldn't we maybe prefer to split the selection of the current folder as its own method that can be called only once as early as we want independently of the setFolders method?

Reply:
It's been revamped with you in #1917.

Reply:
Do we really have to wait for the next PR? Can't we introduce it directly in this PR?

Reply:
Yes, there was too much changes in the next PR to do it in this one.

Reply:
I went and added a comment there